pkg/spanconfig/spanconfig.go

```diff
@@ -40,13 +40,9 @@ type ReconciliationDependencies interface {
 	// applicable to a given tenant.
 	Accessor
```

Comment:
Rename this to KVAccessor, update the comments for all these interfaces to match our API sketch offline. For the Watcher, name it to KVWatcher + rename the package as well, and re-use comments from our sketch. Name the method to WatchForKVUpdates, and WatchForSQLUpdates. Update the comment on spanconfig.Update as well to refer to these new interfaces (was also included in our API sketch offline).